src/App.js

```diff
@@ -1,17 +1,26 @@
 import { useEffect } from "react";
 import { Toaster } from "react-hot-toast";
-import { useSelector } from "react-redux";
+import { useDispatch, useSelector } from "react-redux";
 import { Route, Routes, useNavigate } from "react-router-dom";
 import { PrivateRoute } from "./components";
-import { HomePage, LoginPage, SignupPage } from "./pages";
+import { BookmarkedPage, HomePage, LoginPage, SignupPage } from "./pages";
+import {
+  fetchArchivedPosts,
+  fetchBookmarkedPosts,
+} from "./services/posts/postsService";
 function App() {
   const user = useSelector((state) => state.auth.user);
   const navigate = useNavigate();
+  const dispatch = useDispatch();
   useEffect(() => {
     if (user) {
       navigate("/");
     }
   }, [user]);
+  useEffect(() => {
```

Comment:
You could use the short hand here just a suggestion

Reply:
For if else

Reply:
Okay will do the changes